src/js/jsx/sections/pages/LayerFace.jsx

```diff
@@ -193,15 +229,61 @@ define(function (require, exports, module) {
                 isDropTargetBelow = true;
             }
 
+            var depth = layerStructure.depth(layer),
+                endOfGroupStructure = false,
+                isLastInGroup = false,
+                dragStyle;
+
+            if (isDragTarget) {
+                dragStyle = this.props.dragStyle;
+            } else {
+                // We can skip some rendering calculations if dragging
+                if (layerStructure.byIndex(layerIndex - depth)) {
+                    isLastInGroup = isChildOfSelected &&
+                        layerStructure.byIndex(layerIndex - 1).kind === layer.layerKinds.GROUPEND;
+                }
+
+                // Check to see if this layer is right before the end of nested group
```

Comment:
I don't quite understand what this is supposed to represent. It seems like you just want to know if the visible layer underneath this one in the panel is at the top level? (Or if there is no visible layer below.) Maybe there is a cheaper way to compute this?